### PR TITLE
Update nightly recipe with CLI entrypoint

### PR DESCRIPTION
**Pull request by charlesbluca**
Following up on recent work around the CLI (xref https://github.com/conda-forge/dask-core-feedstock/pull/146), this PR adds the CLI entrypoint to the nightly recipe.

cc @jakirkham 

- [ ] Closes #xxxx
- [ ] Tests added / passed
- [ ] Passes `pre-commit run --all-files`

### CODE REVIEW

File: continuous_integration/recipe/meta.yaml [hunk 1]
```diff
@@ -15,12 +15,13 @@ build:
...
   string: py_{{ GIT_DESCRIBE_HASH }}_{{ GIT_DESCRIBE_NUMBER }}
   script: {{ PYTHON }} -m pip install . -vv
+  entry_points:
+    - dask = dask.__main__:main
 
 requirements:
   host:
     - python >=3.8
     - pip
-    - setuptools
```
Comment (charlesbluca) [inline continuous_integration/recipe/meta.yaml:23]: Unrelated to the CLI changes, but noticed that `setuptools` got dropped as a host dependency from the conda-forge recipe

Comment (jakirkham) [reply continuous_integration/recipe/meta.yaml:23]: Yeah it is a dependency of `pip`. So always gets pulled in

**Review by jakirkham — APPROVED** [review]
LGTM. Thanks Charles! 🙏

**Comment by jakirkham** [comment]
Going to go ahead and merge so we can fix this issue, but we can follow up on anything else as needed

**Review by jrbourbeau — COMMENTED** [review]
Thanks @charlesbluca @jakirkham

**Pull request merged**